src/middlewares/errorMiddleware.js

```
@@ -0,0 +1,16 @@
+import { ApiError } from '../exeptions/api.error.js';
+
+export const errorMiddleware = (err, req, res, next) => {
+  if (err instanceof ApiError) {
+    return res.status(err.status).send({
+      message: err.message,
+      errors: err.errors,
+    });
+  }
+
+  if (err) {
+    return res.status(500).json({ error: 'Server error' });
+  }
```

This catch-all block incorrectly reports all non-`ApiError` types with a generic 500 status. This means your 404 "Not Found" error, which has a status of 404, will be incorrectly reported as a 500 server error. You should check if the `err` object has its own `status` property and use that instead of hardcoding `500`.